File: backend/src/main/java/com/taskflow/kanban/board/service/ColumnService.java
package com.taskflow.kanban.board.service;

import com.taskflow.kanban.board.dto.ColumnCreateDto;
import com.taskflow.kanban.board.dto.ColumnDto;
import com.taskflow.kanban.board.dto.ColumnUpdateDto;

import java.util.List;
import java.util.UUID;

public interface ColumnService {
    ColumnDto createColumn(ColumnCreateDto createDto);
    ColumnDto getColumnById(UUID id);
    List<ColumnDto> getColumnsByBoard(UUID boardId);
    ColumnDto updateColumn(UUID id, ColumnUpdateDto updateDto);
    void deleteColumn(UUID id);
}

File: frontend/src/app/core/services/column.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { BoardColumn, ColumnCreateDto, ColumnUpdateDto } from '../models/board-column.model';

@Injectable({
  providedIn: 'root'
})
export class ColumnService {
  private readonly API_URL = `${environment.apiUrl}/columns`;

  constructor(private http: HttpClient) {}

  getColumnsByBoard(boardId: string): Observable<BoardColumn[]> {
    const params = new HttpParams().set('boardId', boardId);
    return this.http.get<BoardColumn[]>(this.API_URL, { params });
  }

  getColumnById(id: string): Observable<BoardColumn> {
    return this.http.get<BoardColumn>(`${this.API_URL}/${id}`);
  }

  createColumn(column: ColumnCreateDto): Observable<BoardColumn> {
    return this.http.post<BoardColumn>(this.API_URL, column);
  }

  updateColumn(id: string, column: ColumnUpdateDto): Observable<BoardColumn> {
    return this.http.put<BoardColumn>(`${this.API_URL}/${id}`, column);
  }

  deleteColumn(id: string): Observable<void> {
    return this.http.delete<void>(`${this.API_URL}/${id}`);
  }
}
